Feat/on 77 populate case details

tongzxc opened the pull request.

Commit:
update onMounted

farablanco committed:
feat: display case number dynamically

Commit:
feat: populate case details, reference and beneficiary fields

**RELATES TO**: ON-77

Commit:
update requests fields

Commit:
update requests fields

Commit:
feat: populate comments & documents section

**RELATES TO**: ON-77

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-web into feat/on-77-populate-case-details

Commit:
merge from previous on-77

Commit:
feat: add caseDetails.requests and add author to comments section

**RELATES TO**: ON-77

Commit:
feat: update CSS for comments & documents section

**RELATES TO**: ON-77

Commit:
Merge branch 'develop' of https://github.com/TechLadies/onesg-web into feat/on-77-populate-case-details

farablanco commented on src/components/CaseDetails.vue:
```diff
@@ -0,0 +1,368 @@
+<template>
```
what is this file for ?
remove the file if this is not related to  the case details page.

tongzxc replied:
removed file

farablanco commented on src/pages/Cases.vue:
```diff
@@ -171,8 +171,8 @@ ${state.url}`)
       const transformedData = fetchedData.map((data) => {
-        console.log(`typeof:referee`, typeof data.referee)
-        console.log('is it null?', data.referee === null)
+        // console.log(`typeof:referee`, typeof data.referee)
```
just delete this instead of commenting

tongzxc replied:
removed console.log

farablanco commented on src/pages/Details.vue:
```diff
@@ -1,11 +1,16 @@
       <div class="top">
-        <div class="title">Case EF 2020-0104</div>
+        <div class="title">Case #{{caseDetails.caseNumber}}</div>
+        <div v-bind="test" />
```
remove

tongzxc replied:
I don't have this in my file, should be removed already

farablanco commented on src/pages/Details.vue:
Organization

tongzxc replied:
changed to Organisation

farablanco commented on src/pages/Details.vue:
REFERENCE

tongzxc replied:
changed to REFERENCE

farablanco commented on src/pages/Details.vue:
Reference Status

tongzxc replied:
changed to Reference Status

farablanco commented on src/pages/Details.vue:
```diff
@@ -155,7 +188,29 @@
+        </div>
         <br />
       </div>
       <div class="sectionHeadingWhite">NOTES</div>
```
We are missing the section on related cases

tongzxc replied:
updated this section to include related cases

farablanco commented on src/pages/Details.vue:
```diff
@@ -187,42 +244,148 @@ export default {
-    //   // Assign it to Vue data
-    //   table.fetchedItems = transformedData
-    // }
+    // const xx = props
```
remove these

tongzxc replied:
removed

farablanco commented on src/pages/Details.vue:
```diff
@@ -187,42 +244,148 @@ export default {
+    // const xx = props
+    // console.log(props) // { user: '' }
 
+    const refereeStatus = reactive([{//key: the array
```
is this needed?

tongzxc replied:
removed

farablanco commented on src/pages/Details.vue:
```diff
@@ -187,42 +244,148 @@ export default {
 
+    const refereeStatus = reactive([{//key: the array
+    }])
+    const amountGranted = reactive([{//key: the array
```
is this needed?

tongzxc replied:
removed

farablanco commented on src/pages/Details.vue:
Amount granted and reference status do  not reflect value from DB
<img width="1209" alt="Screenshot 2021-03-13 at 6 45 23 PM" src="https://user-images.githubusercontent.com/28174698/111027684-c5fc3600-842c-11eb-975d-d458e3482842.png">

tongzxc replied:
have updated to show status and amount granted from db

farablanco reviewed (changes requested):
Hi. I’ve completed the review. Almost there. I have a couple of comments. The most important ones would be on missing UI fields and UI fields not reflecting info from the DB. Please check comments and update.

Commit:
feat: update referee status section

drop down to show reference status
amount granted to show the amount granted, if not show 0, and show S$ before the value

**RELATES TO**: ON-77

Commit:
change referee to reference in headers

Commit:
feat: added related cases

retrieve related case numbers from /v1/beneficiaries/:id/cases
populate header to show number of related cases
added links to the respective related case detail url
update CSS for related case links

**RELATES TO**: ON-77

Commit:
remove console.log from related cases section

Commit:
update based on PR comments

Commit:
remove Checkbox

Commit:
feat: update PR

update comments endpoint
update comments section's dynamic author and date

**RELATES TO**: ON-77

Commit:
feat: include route for related cases

added :key="$route.fullPath" in Secure.vue and Public.vue
create goToCase function

**RELATES TO**: ON-77

Commit:
remove console.log

Commit:
removed div v-bind="test" />

Commit:
remove console.logs

Commit:
feat: update PR

include relatedCases.length in template
remove paymentType and include directly to string
added comments

**RELATES TO**: ON-77

farablanco committed:
chore: add refactor notes

farablanco reviewed (commented):
added a refactor suggestion.
This can be picked up by whoever is going to work on the project next
